pkg/util/kubernetes/apiserver/openshift.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+import (
+	"errors"
+
+	osq "github.com/openshift/api/quota/v1"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var ErrNotOpenShift = errors.New("not an OpenShift cluster")
+
+// IsOpenShift detects available endpoints to determine of OpenShift APIs are available
+func (c *APIClient) IsOpenShift() OpenShiftApiLevel {
+	if c.isOpenShift != OpenShiftUnknown {
+		return c.isOpenShift
+	}
+	err := c.Cl.CoreV1().RESTClient().Get().AbsPath("/apis/quota.openshift.io").Do().Error()
+	if err == nil {
+		c.isOpenShift = OpenShiftAPIGroup
+		return c.isOpenShift
+	}
+	log.Debugf("Cannot access new OpenShift API: %s", err)
+
+	err = c.Cl.CoreV1().RESTClient().Get().AbsPath("/oapi").Do().Error()
+	if err == nil {
+		c.isOpenShift = OpenShiftOApi
+		return c.isOpenShift
+	}
+	log.Debugf("Cannot access old OpenShift OAPI: %s", err)
+
+	// Fallback to NotOpenShift
+	c.isOpenShift = NotOpenShift
+	return c.isOpenShift
+}
+
+// ListOShiftClusterQuotas retrieves Openshift ClusterResourceQuota objects
+// from the APIserver, returns ErrNotOpenShift if called on a non-Openshift cluster
+func (c *APIClient) ListOShiftClusterQuotas() ([]osq.ClusterResourceQuota, error) {
+	var url string
+	switch c.IsOpenShift() {
```

Comment:
ListOShiftClusterQuotas is called after evaluating IsOpenShift(), is there any way we can factorize ?
Trying to reduce the number of calls to the API Server

Reply:
I do agree `c.isOpenShift` is not very elegant. WDYT about keeping this in the check itself, and even move `ListOShiftClusterQuotas` there, as I don't foresee other uses for it? `GetRESTObject` would stay under apiclient, as it can have other future uses.

Reply:
I am a fan of having a dedicated openshift.go file, I think all the relevant methods should be there. 👍 for the `GetRESTObject` under apiClient. If there are other things to list in the next versions of OpenShift, they could go alongside `ListOShiftClusterQuotas`.
Would that make sense ?